v-next/hardhat-typechain/src/internal/generate-types.ts

```diff
@@ -140,3 +142,37 @@ export function addJsExtensionsIfNeeded(content: string): string {
       `import ${imports} from ${quote}${path}/index.js${quote};`,
   );
 }
+
+function addSupportForAttachMethod(modifiedContent: string): string {
+  const pattern = /class\s+(\w+)__factory/; // Pattern to find the contract name in factory files
```

Comment:
Just a note for other reviewers because I checked it myself, `\w+` is equivalent to ` [a-zA-Z0-9_]+`. So it should cover all the characters allowed in contract names 👍